### PR TITLE
Add support for docutils 0.17

### LINKED ISSUE #1115: rendering wrong in docutils 0.17 (unordered list, maybe more?)
I am giving this issue a broad title to collect other issues.

### Problem

With docutils 0.17, it seems that bullet points don't render properly (the marker is missing), spacing too little (see images below)

An immediate fix is already in sphinx_rtd_theme==0.5.2 (just released), see (#1111, #1112 for issues, #1114 for fix), but that seems to only pin docutils==0.16 so other things that broke should still be tracked.  I tested using the sphinx_rtd_theme docs themselves.  #1112 shows this with a problem in the sidebar.

With 0.16:
> image

With 0.17:
> image

I originally thought this was another Sphinx problem, but it seems to not be present in the default Sphinx theme, but (only/at least) sphinx_rtd_theme.

### Reproducible Project

sphinx_rtd_theme with docutils==0.16.  Installed from the master branch `pip install docutils=0.17` immediately after.
```
git clone https://github.com/readthedocs/sphinx_rtd_theme
cd sphinx_rtd_theme
git checkout ca2719b
pip install -e . -r docs/requirements.txt docutils==0.17
(cd docs ; make clean html)
# open build/html/changelog.html, does not work
pip install docutils==0.16
# refresh page, works
```

#### Error Logs/Results

none I see as relevant

#### Expected Results

...

### Environment Info

- Python Version: Python 3.7.3
- Sphinx Version: Sphinx                        3.5.3
- RTD Theme Version: ca2719b

```
Package                       Version   Location                          
----------------------------- --------- ----------------------------------
alabaster                     0.7.12    
Babel                         2.9.0     
certifi                       2020.12.5 
chardet                       4.0.0     
docutils                      0.16      
idna                          2.10      
imagesize                     1.2.0     
Jinja2                        2.11.3    
MarkupSafe                    1.1.1     
packaging                     20.9      
pip                           18.1      
pkg-resources                 0.0.0     
Pygments                      2.8.1     
pyparsing                     2.4.7     
pytz                          2021.1    
requests                      2.25.1    
setuptools                    40.8.0    
six                           1.15.0    
snowballstemmer               2.1.0     
Sphinx                        3.5.3     
sphinx-rtd-theme              0.5.2     /home/$USER/git/sphinx_rtd_theme
sphinxcontrib-applehelp       1.0.2     
sphinxcontrib-devhelp         1.0.2     
sphinxcontrib-htmlhelp        1.0.3     
sphinxcontrib-httpdomain      1.7.0     
sphinxcontrib-jsmath          1.0.1     
sphinxcontrib-qthelp          1.0.3     
sphinxcontrib-serializinghtml 1.1.4     
urllib3                       1.26.4    
```

# See also
- Docutils 0.17 changelog: https://docutils.sourceforge.io/RELEASE-NOTES.html#id46 (lists changes in outputs)
- https://github.com/sphinx-doc/sphinx/issues/9061 (sidebar again, looks like sphinx_rtd_theme in that example).  I first reported this issue there but it could'nt be reproduced, seems to be sphinx_rtd_theme but not the default theme
- Plus some more PRs around sphinx-doc/sphinx

### PR DESCRIPTION
**Pull request by Blendify**
This should'nt be a breaking change as backward compatibility is preserved so this change could go into 1.0 or wait until 1.1.

I know a lot of users are asking for this so it might be nice to include this in 1.0.

Fixes #1115

### CODE REVIEW

File: src/sass/_theme_rst.sass [hunk 1]
```diff
@@ -16,22 +16,18 @@
 .rst-content
   // Sphinx by default applies HxW style attributes to images. This fixes that oversite.
   img
+    margin-bottom: $base-line-height
```
Comment (agjohnson) [inline src/sass/_theme_rst.sass:19]: I'm assuming there will probably be unwanted side effects from this, `img` is a pretty generic selector. Is this on purpose? Looks like the original selector was a bit more specific, at `.rst-content .section > img`

**Review by agjohnson — COMMENTED** [review]
Nice! Glad to have this fixed. I'll note a few visual differences I noticed in QA

**Comment by agjohnson** [comment]
Gave a quick test against Sphinx 4.1 + `master` branch on Firefox. Lists look far better!

----

There is some probably unwanted spacing added with this PR, under the image in this hlist:

On master:

image

On this PR:

image

----

image

----

Some unwanted spacing under an inline image:

image

----

More spacing was added here, but it actually looks better. If we want to undo the image spacing in this PR, I would make a separate PR with this change included:

image

Same with image spacing and this caption:

image

----

Probably a bit too much spacing here now:

image

**Comment by MehdiChinoune** [comment]
Is there an issue with `docutils>=0.18`

**Pull request merged**